### PR TITLE
audio text to text task guide

### PR DESCRIPTION
**Pull request by Deep-unlearning**
### What does this PR do?

Adds a new task guide for audio-text-to-text models, covering how to fine-tune Voxtral for audio reasoning tasks using LoRA.

### CODE REVIEW

File: docs/source/en/tasks/audio_text_to_text.md [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Audio-text-to-text
+
+[[open-in-colab]]
+
+Audio-text-to-text models, are language models that take audio and text inputs and generate text outputs. These models combine audio understanding with language generation capabilities, enabling tasks like audio question answering, audio reasoning, and spoken dialogue understanding. Unlike traditional ASR models that simply transcribe speech, audio-text-to-text models can reason about audio content, follow instructions, and generate contextual responses.
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]:
```suggestion
Audio-text-to-text models, take audio and text inputs and generate text outputs. These models can understand audio and generate text, enabling tasks like audio question answering, audio reasoning, and spoken dialogue understanding. Unlike traditional ASR models that simply transcribe speech, audio-text-to-text models can reason about audio content, follow instructions, and generate contextual responses.
```

File: docs/source/en/tasks/audio_text_to_text.md [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md:22]: I think it's better if we show simple inference here with a model and then get to fine-tuning, I find it more intuitive

File: docs/source/en/tasks/audio_text_to_text.md [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+This guide will show you how to:
+
+1. Fine-tune [Voxtral](https://huggingface.co/mistralai/Voxtral-Mini-3B-2507) on the [Big Bench Audio](https://huggingface.co/datasets/ArtificialAnalysis/big_bench_audio) dataset for audio reasoning tasks using LoRA.
+2. Use your fine-tuned model for inference.
+
+<Tip>
+
+To see all architectures and checkpoints compatible with this task, we recommend checking the [task-page](https://huggingface.co/tasks/audio-text-to-text)
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]:
```suggestion
To see all architectures and checkpoints compatible with this task, we recommend checking the [task-page](https://huggingface.co/tasks/audio-text-to-text).
```

File: docs/source/en/tasks/audio_text_to_text.md [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+```py
+>>> from datasets import load_dataset, Audio
+
+>>> dataset = load_dataset("ArtificialAnalysis/big_bench_audio", split="train")
+```
+
+Cast the audio column to 16kHz, which is required by Voxtral:
+
+```py
+>>> dataset = dataset.cast_column("audio", Audio(sampling_rate=16000))
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md:90]: is it GPU poor friendly?

File: docs/source/en/tasks/audio_text_to_text.md [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+...         
+...         # Process conversations
+...         inputs = self.processor.apply_chat_template(
+...             batch_conversations,
+...             tokenize=True,
+...             return_tensors="pt",
+...             padding=True,
+...         )
+...         
+...         prompt_ids = inputs["input_ids"]
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: interesting that we need to split these keys and pass to tokenizer separately, so processor doesn't directly take in like `processor(**inputs)`? if so imo UX can be improved

File: docs/source/en/tasks/audio_text_to_text.md [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+...             non_pad_len = sum(p_att)
+...             p_ids = p_ids[:non_pad_len]
+...             p_att = p_att[:non_pad_len]
+...             
+...             # Concatenate: prompt + answer + eos
+...             ids = p_ids + a_ids + [tok.eos_token_id]
+...             attn = p_att + [1] * (len(a_ids) + 1)
+...             
+...             # Labels: mask prompt (-100), learn only answer
+...             lab = [-100] * len(p_ids) + a_ids + [tok.eos_token_id]
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: this should def be handled in processor, we should ask audio folks imo

**Review by merveenoyan — COMMENTED** [review]
left very general comments!

File: docs/source/en/tasks/audio_text_to_text.md [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+[[open-in-colab]]
+
+Audio-text-to-text models, are language models that take audio and text inputs and generate text outputs. These models combine audio understanding with language generation capabilities, enabling tasks like audio question answering, audio reasoning, and spoken dialogue understanding. Unlike traditional ASR models that simply transcribe speech, audio-text-to-text models can reason about audio content, follow instructions, and generate contextual responses.
+
+This guide will show you how to:
+
+1. Fine-tune [Voxtral](https://huggingface.co/mistralai/Voxtral-Mini-3B-2507) on the [Big Bench Audio](https://huggingface.co/datasets/ArtificialAnalysis/big_bench_audio) dataset for audio reasoning tasks using LoRA.
+2. Use your fine-tuned model for inference.
+
+<Tip>
```
Comment (stevhliu) [inline docs/source/en/tasks/audio_text_to_text.md]: we've started using this syntax now as its simpler. let's replace all instances of `<Tip></Tip>` with this :)

```suggestion
> [!TIP]
```

File: docs/source/en/tasks/audio_text_to_text.md [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+
+```py
+>>> data_collator = VoxtralAudioCollator(processor)
+```
+
+## Train
+
+<Tip>
+
+If you aren't familiar with finetuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training#train-with-pytorch-trainer)!
```
Comment (stevhliu) [inline docs/source/en/tasks/audio_text_to_text.md]:
```suggestion
If you aren't familiar with finetuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training)!
```

**Review by stevhliu — COMMENTED** [review]
thanks, just a couple minor nits from me!

File: docs/source/en/tasks/audio_text_to_text.md (hunk 7 above)
Comment (Deep-unlearning) [reply docs/source/en/tasks/audio_text_to_text.md]: thanks!

File: docs/source/en/tasks/audio_text_to_text.md [hunk 9]
```diff
@@ -0,0 +1,380 @@
...
+    tokenize=True,
+    add_generation_prompt=True,
+    return_dict=True,
+).to(model.device)
+
+outputs = model.generate(**inputs, max_new_tokens=500)
+
+decoded_outputs = processor.batch_decode(outputs[:, inputs.input_ids.shape[1]:], skip_special_tokens=True)
+print(decoded_outputs)
+```
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: let's print the output?

File: docs/source/en/tasks/audio_text_to_text.md [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+
+This guide will show you how to:
+
+1. Fine-tune [Audio Flamingo 3](https://huggingface.co/nvidia/audio-flamingo-3-hf) on the [AudioCaps](https://huggingface.co/datasets/OpenSound/AudioCaps) dataset for audio captioning using LoRA.
+2. Use your fine-tuned model for inference.
+
+> [!TIP]
+
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: text below this newline breaks, let's remove newline

File: docs/source/en/tasks/audio_text_to_text.md [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+To see all architectures and checkpoints compatible with this task, we recommend checking the [task-page](https://huggingface.co/tasks/audio-text-to-text).
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]:
```suggestion
> To see all architectures and checkpoints compatible with this task, we recommend checking the [task-page](https://huggingface.co/tasks/audio-text-to-text).
```
for tip block

File: docs/source/en/tasks/audio_text_to_text.md [hunk 12]
```diff
@@ -0,0 +1,380 @@
...
+
+```bash
+pip install transformers datasets peft accelerate
+```
+
+We encourage you to login to your Hugging Face account so you can upload and share your model with the community. When prompted, enter your token to login:
+
+```py
+>>> from huggingface_hub import notebook_login
+
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: let's remove newlines in codeblocks

File: docs/source/en/tasks/audio_text_to_text.md [hunk 13]
```diff
@@ -0,0 +1,380 @@
...
+Start by loading the [AudioCaps](https://huggingface.co/datasets/OpenSound/AudioCaps) dataset from the 🤗 Datasets library in streaming mode. This dataset contains audio clips with descriptive captions, perfect for audio captioning tasks.
+
+```py
+>>> from datasets import load_dataset, Audio
+
+>>> dataset = load_dataset("OpenSound/AudioCaps", split="train", streaming=True)
+```
+
+Cast the audio column to 16kHz, which is required by Audio Flamingo's Whisper feature extractor:
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md:87]: very nice!

File: docs/source/en/tasks/audio_text_to_text.md [hunk 14]
```diff
@@ -0,0 +1,380 @@
...
+
+```py
+>>> data_collator = AudioFlamingo3DataCollator(processor)
+```
+
+## Train
+
+> [!TIP]
+
+If you aren't familiar with finetuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training)!
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]:
```suggestion
> If you aren't familiar with fine-tuning a model with the [`Trainer`], take a look at the basic tutorial [here](../training)!
```

File: docs/source/en/tasks/audio_text_to_text.md [hunk 15]
```diff
@@ -0,0 +1,380 @@
...
+Instantiate the data collator:
+
+```py
+>>> data_collator = AudioFlamingo3DataCollator(processor)
+```
+
+## Train
+
+> [!TIP]
+
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: newline warning

File: docs/source/en/tasks/audio_text_to_text.md [hunk 16]
```diff
@@ -0,0 +1,380 @@
...
+...     lora_dropout=0.05,
+...     bias="none",
+...     task_type="CAUSAL_LM",
+... )
+
+>>> model = get_peft_model(model, lora_config)
+>>> model.print_trainable_parameters()
+```
+
+> [!TIP]
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md:216]: same here

File: docs/source/en/tasks/audio_text_to_text.md [hunk 17]
```diff
@@ -0,0 +1,380 @@
...
+
+This LoRA configuration targets the language model's attention and feed-forward layers, which is sufficient for fine-tuning on audio captioning tasks. The audio encoder remains frozen and uses its pretrained weights.
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: you could say that it's to save on compute etc?

File: docs/source/en/tasks/audio_text_to_text.md [hunk 18]
```diff
@@ -0,0 +1,380 @@
...
+```py
+>>> trainer.save_model()
+>>> processor.save_pretrained("audio-flamingo-3-hf-lora-finetuned")
+```
+
+Once training is completed, you can optionally share your model to the Hub:
+
+```py
+>>> # Uncomment to push to Hub
+>>> # trainer.push_to_hub()
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: let's push by default!

**Review by merveenoyan — COMMENTED** [review]
very nice! just left some nits, we can merge after that

File: docs/source/en/tasks/audio_text_to_text.md [hunk 19]
```diff
@@ -0,0 +1,364 @@
...
+
+-->
+
+# Audio-text-to-text
+
+[[open-in-colab]]
+
+Audio-text-to-text models, take audio and text inputs and generate text outputs. These models can understand audio and generate text, enabling tasks like audio question answering, audio reasoning, and spoken dialogue understanding. Unlike traditional ASR models that simply transcribe speech, audio-text-to-text models can reason about audio content, follow instructions, and generate contextual responses.
+
+Here is an example of how to use an audio-text-to-text model:
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: let's put a bit more explanation on inference example (assume less)

File: docs/source/en/tasks/audio_text_to_text.md [hunk 20]
```diff
@@ -0,0 +1,364 @@
...
+...                     ],
+...                 },
+...                 {
+...                     "role": "assistant",
+...                     "content": [{"type": "text", "text": feature["caption"]}],
+...                 }
+...             ]
+...             conversations.append(sample)
+...
+...         # Apply chat template with automatic label generation
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md]: can we rephrase this for clarity? when you say generation it sounds like _generation_ in the sense of model

File: docs/source/en/tasks/audio_text_to_text.md [hunk 21]
```diff
@@ -0,0 +1,364 @@
...
+
+Generate a response:
+
+```py
+>>> with torch.no_grad():
+...     output_ids = model.generate(**inputs, max_new_tokens=100)
+>>> # Decode only the generated tokens
+>>> input_len = inputs["input_ids"].shape[1]
+>>> response = processor.tokenizer.decode(output_ids[0][input_len:], skip_special_tokens=True)
+>>> print(response)
```
Comment (merveenoyan) [inline docs/source/en/tasks/audio_text_to_text.md:330]: can we print the output?

**Review by merveenoyan — APPROVED** [review]
left three minor comments, lgtm afterwards!
